src/WebJobs.Extensions.DurableTask/DurableTaskJobHostConfigurationExtensions.cs

```diff
@@ -107,6 +111,22 @@ public static IWebJobsBuilder AddDurableTask(this IWebJobsBuilder builder)
             return builder;
         }
 
+#if FUNCTIONS_V3_OR_GREATER
+        /// <summary>
+        /// Adds the <see cref="IScaleMonitor"/> and <see cref="ITargetScaler"/> providers for the Durable Triggers.
+        /// </summary>
+        /// <param name="builder">The <see cref="IWebJobsBuilder"/> to configure.</param>
+        /// <returns>Returns the provided <see cref="IWebJobsBuilder"/>.</returns>
+        internal static IWebJobsBuilder AddDurableScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
+        {
+            // this segment adheres to the followings pattern: https://github.com/Azure/azure-sdk-for-net/pull/38673/files
+            builder.Services.AddSingleton(serviceProvider => new DurableTaskTriggersScaleProvider(serviceProvider.GetService<IOptions<DurableTaskOptions>>(), serviceProvider.GetService<INameResolver>(), serviceProvider.GetService<ILoggerFactory>(), serviceProvider.GetService<IEnumerable<IDurabilityProviderFactory>>(), triggerMetadata));
```

Comment:
We can pass serviceProvider

Reply:
Thanks @alrod. I think the code did this originally, but as per this comment we decided not to resolve services in the constructor: https://github.com/Azure/azure-functions-durable-extension/pull/2462#discussion_r1336409281